### ISSUE TITLE
Zenodo GitHub app requires write admin access to the organization

### ISSUE THREAD

**Issue by mrybczyn** [zenodo/zenodo#2453]
Browser & OS (see also https://www.whatismybrowser.com/): Not applicable

## Describe the bug
When installing the GitHub app requires write admin permissions. This is a security concern and a potential attack vector on the application source repository.

What does the app require? If it is just a webhook or a similar change, this could be done manually before installing the app.

## Steps to Reproduce
1. Start installing the app
2. See the permission request

## Expected behavior

No write admin permissions are required (ideally, no admin permissions at all)

## Screenshots (if applicable)

NA

## Additional context

We have been asked to enable Zenodo for one of the Eclipse Foundation's projects. However, per our policy, access to the admin permissions is limited, and developers do not have it.

**Comment by fmigneault** [comment]
I'm also concerned by the same permissions. 

Admin access to webhooks only to connect the GitHub account to the profile! Nope. That is way too much power and can give access to other critical webhook configurations defined in my account. 

Many other websites allow GitHub integration with only read access to some profile information simply to validate the user. 
What mandates the use admin webhooks? 

See also #1899 that asked about the same problem in 2019. 
Is there any plan to address this at all, or at the very least provide an answer?

**Comment by r0sek** [comment]
I also think my organization will not grant Zenodo permissions, and can't find instructions for how to do this manually.

**Comment by eeholmes** [comment]
My organization will also not grant Zenodo admin access to webhooks. If there was a way for me to manually add repos on the GitHub page on Zenodo, I would not have to grant this access. But there is no way for me to add a url to my repo on the organization.

**Comment by r0sek** [comment]
@eeholmes I also contacted info@zenodo.org about this and got this response: 

> If you want your dataset to be citable with a Zenodo DOI, the dataset will need to be stored in Zenodo.
This is also documented in the official GitHub documentation: https://docs.github.com/en/repositories/archiving-a-github-repository/referencing-and-citing-content

> The main reason is that Git history can be rewritten, and repositories can be deleted.
> A repository like Zenodo guarantees that once a DOI is issued for a dataset stored in Zenodo, the data will not change.
> We actually found out that many GitHub repositories archived on Zenodo do not exist anymore on Github.

I assumed that I would just need to register the URL of the GitHub or a directory within it or something like that, but it now makes sense that I need to upload and archive the actual file. I haven't done this yet, but I believe you would just log in to Zenodo and [create a new upload](https://zenodo.org/uploads/new).

**Comment by eeholmes** [comment]
@r0sek Yes I can always upload manually. But I'd like to be able to create a webhook so that the archiving is triggered automatically. If you link a repo in your personal GitHub account, you can grant Zenodo access to create a webhook. Then link a repository. Now you can go to that repository settings and you will see on the left the nav to go to webhooks. Go to the webhook for Zenodo. It looks simple but I need an access token. I am not sure how to get that.  Maybe this? https://zenodo.org/account/settings/applications/tokens/new/

Currently we have a silly workaround. Create the repo in your personal account. Link to Zenodo. Transfer the repo into the organization. The webhook still works it seems.

Update: the workaround doesn't work. The moment you sync the GitHub repos (on GitHub page of Zenodo) is stops seeing the moved repo and removes it from the Enable Repositories list and your get an error "repo does not have webhooks enabled"

**Comment by eeholmes** [comment]
See also this for another solution without needing admin access to webhooks: https://github.com/rseng/zenodo-release

**Comment by fmigneault** [comment]
The GitHub repositories we are trying to link are all public. There is no need for admin level access to obtain the archive. Anyone can download this archive from the released versions, which are the same pointers to be linked by Zenodo. 

The whole point of adding the Zenodo app is to automate the process. Doing the step manually (as we currently do) is what we are trying to address.

@eeholmes Thanks for the reference. [rseng/zenodo-release](https://github.com/rseng/zenodo-release) looks like a promising workaround. It would be great if the official app could do something similar.
